plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/table/HudiTableFileSystemView.java

```diff
@@ -293,9 +292,7 @@ private FileIterator listPartition(Location partitionLocation)
         if (fileIterator.hasNext()) {
             return fileIterator;
         }
-        try (OutputStream ignored = metaClient.getFileSystem().newOutputFile(partitionLocation).create()) {
-            return FileIterator.empty();
-        }
+        return FileIterator.empty();
```

Comment:
This change is unrelated to the fix of query hang, right? Please separate a commit if so. The added test passes without this change as far as I confirmed.

Reply:
it is part of this commit only it is related

Reply:
this is the place it got hanged